zerver/views/video_calls.py

```diff
@@ -161,18 +161,29 @@ def deauthorize_zoom_user(request: HttpRequest) -> HttpResponse:
     return json_success()
 
 
-def get_bigbluebutton_url(request: HttpRequest, user_profile: UserProfile) -> HttpResponse:
+@has_request_variables
+def get_bigbluebutton_url(request: HttpRequest, user_profile: UserProfile,
+                          meeting_name: str = REQ()) -> HttpResponse:
     # https://docs.bigbluebutton.org/dev/api.html#create for reference on the api calls
     # https://docs.bigbluebutton.org/dev/api.html#usage for reference for checksum
     id = "zulip-" + str(random.randint(100000000000, 999999999999))
     password = b32encode(secrets.token_bytes(7))[:10].decode()
-    checksum = hashlib.sha1(("create" + "meetingID=" + id + "&moderatorPW="
-                             + password + "&attendeePW=" + password + "a" + settings.BIG_BLUE_BUTTON_SECRET).encode()).hexdigest()
-    url = add_query_to_redirect_url("/calls/bigbluebutton/join", urlencode({
-        "meeting_id": "\"" + id + "\"",
-        "password": "\"" + password + "\"",
-        "checksum": "\"" + checksum + "\""
-    }))
+    create_params = urlencode({  # type: ignore[type-var] # https://github.com/python/typeshed/issues/4234
+        "meetingID": id,
+        "name": meeting_name,
+        "moderatorPW": password,
+        # We generate the password this way to make sure only the server knows it.
+        # This is a easy way to prevent the user to get the data of the shared url
+        # and use it later and skipping Zulip validation
```

Comment:
I don't exactly understand what kind of attack scenario is being described here. Could you explain in a bit more detail, that'll make this clearer for someone who's not familiar with bluebutton?

Reply:
The API of BigBlueButton does require a checksum to be checksum to match with the parameters of the request.
We generate this checksum on generation of the BigBlueButton Link, so that the parameters (e.g. Id of the Meeting, Password of the Meeting, ...) can not be manipulated by the user.
By doing this we fix the issue of manipulating the parameters, but open the possibility of a user, after no longer being part of the Zulip Organization, having saved a link and passing the parameters directly to the Big Blue Button API and authenticating with the checksum.
By generating the attendeePW this way, only the server knows the correct parameter for attendeePW and thus only the server can make the API request, as the checksum will fail with other values.

I think the better solution would be to use an own checksum, which Zulip validates and generates the checksum for Big Blue Button afterwards. (After clicking the link). That way the Client would never see the Big Blue Button checksum for the create call.

Furthermore I admit that the risk involved from a user being able to create the API call is really minimal, as this would not grant the user permission to enter the Meeting, but only create it. (After the createTime addition)

Reply:
changed to a better solution in `verify url data` commit